### ISSUE TITLE
Warn user if unable to read/write their user preferences

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#12974]
Currently (in Electron, Qt desktop, and Server), if the user preferences can't be written (e.g. file permissions are messed up), a message is shown:

<img width="634" alt="Screenshot 2023-04-13 at 12 07 29" src="https://user-images.githubusercontent.com/10569626/231846563-f2e7d445-22e3-4b91-b6d2-a3f3312fe2ff.png">

We could show something a bit more helpful, such as saying which file we can't access:

`~/.config/rstudio/rstudio-prefs.json` on Mac and Linux
`%appdata%\RStudio\rstudio-prefs.json` on Windows

**Comment by gtritchie** [comment]
@ronblum Hmm, this isn't doing what I saw when I opened it.

I made my preferences file read-only to myself, and when I tried saving a preference got this (both with Electron and Server):

<img width="374" alt="Screenshot of error dialog saying Unable to save preferences" src="https://github.com/rstudio/rstudio/assets/10569626/43226da6-2939-473d-a034-7576bddc9c98">

This message was added 3+ years ago via https://github.com/rstudio/rstudio/pull/7742.

Given that, unless I'm missing something, we can probably close this.
